client/ayon_core/pipeline/load/utils.py

```diff
@@ -671,139 +674,227 @@ def get_representation_path_with_anatomy(repre_entity, anatomy):
         anatomy (Anatomy): Project anatomy object.
 
     Returns:
-        Union[None, TemplateResult]: None if path can't be received
+        TemplateResult: Resolved representation path.
 
     Raises:
         InvalidRepresentationContext: When representation data are probably
             invalid or not available.
+
     """
+    return get_representation_path(
+        anatomy.project_name,
+        repre_entity,
+        anatomy=anatomy,
+    )
 
-    try:
-        template = repre_entity["attrib"]["template"]
 
+def get_representation_path_with_roots(
+    representation: dict[str, Any],
+    roots: dict[str, str],
+) -> Optional[TemplateResult]:
+    """Get filename from representation with custom root.
+
+    Args:
+        representation(dict): Representation entity.
+        roots (dict[str, str]): Roots to use.
+
+
+    Returns:
+        Optional[TemplateResult]: Resolved representation path.
+
+    """
+    try:
+        template = representation["attrib"]["template"]
     except KeyError:
-        raise InvalidRepresentationContext((
-            "Representation document does not"
-            " contain template in data ('data.template')"
-        ))
+        return None
 
     try:
-        context = repre_entity["context"]
-        _fix_representation_context_compatibility(context)
-        context["root"] = anatomy.roots
+        context = representation["context"]
 
-        path = StringTemplate.format_strict_template(template, context)
+        _fix_representation_context_compatibility(context)
 
-    except TemplateUnsolved as exc:
-        raise InvalidRepresentationContext((
-            "Couldn't resolve representation template with available data."
-            " Reason: {}".format(str(exc))
-        ))
+        context["root"] = roots
+        path = StringTemplate.format_strict_template(
+            template, context
+        )
+    except (TemplateUnsolved, KeyError):
+        # Template references unavailable data
+        return None
 
     return path.normalized()
 
 
-def get_representation_path(representation, root=None):
-    """Get filename from representation document
+def _get_representation_path_decorator(func):
```

Comment:
I think the name shouldn't have `_decorator` in it as the usage `@_get_representation_path_decorator` implies it IS a decorator. It is also not very obvious from the use what it does:

```python
@_get_representation_path_decorator
def get_representation_path(..,):
    ...
```

and the docstring of the decorator doesn't tell much about what it does. What about `upgrade_representation_api` or something like that